src/repositories/soft-crud.repository.base.ts

```diff
@@ -157,6 +170,24 @@ export abstract class SoftCrudRepository<
     }
   }
 
+  //find by Id including soft deleted record
+  async findByIdIncludeSoftDelete(
+    id: ID,
+    filter?: Filter<T>,
```

Comment:
no need of filter here. It doesnot make sense.

Reply:
if anyone wants to use a filter here like a filter excluding where

Reply:
Not possible in findbyid

Reply:
loopback itself gives in its default crud repository something like this **findById(id: ID, filter?: FilterExcludingWhere<T>, options?: Options): Promise<T & Relations>;** here is the link [](https://loopback.io/doc/en/lb4/apidocs.repository.defaultcrudrepository.findbyid.html) and in example too , loopback explains like 
async findById(
    @param.path.number('id') id: number,
    @param.filter(Todo, {exclude: 'where'}) filter?: FilterExcludingWhere<Todo>
  ): Promise<Todo> {
    return this.todoRepository.findById(id, filter);
  }

  @patch('/todos/{id}', {
    responses: {
      '204': {
        description: 'Todo PATCH success',
      },
    },
  })
here is the link to an example resource [](https://loopback.io/doc/en/lb4/Controller-generator.html)

Reply:
Ah ! Now I remember why they have it. Its for supporting relations. Makes sense.